### PR TITLE
refactor: declare loop variables locally in for loops

### PR DESCRIPTION
**Pull request by jhillacre**
Ensure all loop variables are declared with `var` inside `for` loops to fix scoping issues and prevent implied globals.

### CODE REVIEW

**Comment by jhillacre** [comment]
This is bound to create conflicts, any large overhaul branches should probably merge first.

It'll be easier to handle conflicts on this side.

File: scripts/explode_script/explode_script.gml [hunk 1]
```diff
@@ -9,7 +9,7 @@ function explode_script(argument0, argument1) {
 	  explode[0]=0;
 	  explode[1]=0;
 	}
-	for (i=0; i<string_count(argument1,argument0); i+=1)
+	for (var i=0; i<string_count(argument1,argument0); i+=1)
```
Comment (OH296) [inline scripts/explode_script/explode_script.gml]:
```suggestion
	for (var i=0; i<string_count(argument1,argument0); i++)
```

File: scripts/scr_imperial_navy_functions/scr_imperial_navy_functions.gml [hunk 2]
```diff
@@ -193,7 +193,7 @@ function navy_attack_player_world(){
 	    if (instance_exists(orbiting)){
 	        var tar=0;
 			var i=0;
-	        for (var i=1;i<=orniting.planets;i++){
+	        for (i=1;i<=orniting.planets;i++){
```
Comment (OH296) [inline scripts/scr_imperial_navy_functions/scr_imperial_navy_functions.gml]: is `orniting` a spelling error in this instance?

**Review by OH296 — APPROVED** [review]
Yeah i'm happy with this, it used to be like this and then someone went through and had a rant changed it as their only commit and promptly left, oh well live and learn.

**Comment by OH296** [comment]
@EttyKitty not aware there are many big overhaul branches that can't merge and work around this now, afaik the new sprite and the chapter save were the big ones so once they're fully merged (i think they already are) probably get this in ASAP

File: scripts/scr_imperial_navy_functions/scr_imperial_navy_functions.gml (hunk 2 above)
Comment (jhillacre) [inline scripts/scr_imperial_navy_functions/scr_imperial_navy_functions.gml]: Amazing this hasn't generated an error report. Makes me wonder if it is used.

```suggestion
	        for (i = 1; i <= orbiting.planets; i++) {
```

**Comment by jhillacre** [comment]
> @EttyKitty not aware there are many big overhaul branches that can't merge and work around this now, afaik the new sprite and the chapter save were the big ones so once they're fully merged (i think they already are) probably get this in ASAP

Still update to date with base, merge away in that case

**Pull request merged**